streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableAggregate.java

```diff
@@ -47,8 +47,10 @@
     }
 
     @Override
-    public void enableSendingOldValues() {
+    public boolean enableSendingOldValues(final boolean onlyIfMaterialized) {
```

Comment:
Can we flip the logic of the boolean flag ? -- It makes a knob in my mind to say, `onlyIfMaterailized=false` implies that we need to (ie, enforce to) materialize...  I would prefer the suggest from Guozhang to use a flag like `enforceMaterialization` (or maybe better `materializeIfNeeded`).

Also, the method may or may not enable sending old values and thus, we might want to rename it to `maybeEnableSendOldValues` (or `enableSendOldValuesIfPossible` or similar).

Reply:
I did initially go with this. However, it seems really unintuitive. 

Given a param such as `forceMaterialization` then it's clear that `enableSendingOldValues(true)` will force materialization, but what does `enabledSendingOldValues(false)` mean? OK, we're not forcing materialization, but actually the method won't enable sending old values if the param is `false` and its not already materialized.

Even if we go with `maybeEnableSendingOldValues(enforceMaterialization)` I still would argue the semantics are not intuitive from the names alone.  (Though of course JavaDocs would help).

However, given `enableSendingOldValues(onlyIfMaterialized)` is, IMHO, very intuitive. `enableSendingOldValues(true)` will only enable sending old values if already materialized, whereas `enableSendingOldValues(false)` will always enable sending old values, materializing as necessary, must like the previous `enableSendingOldValues()` method did.

Likewise, if we stick with `enableSendingOldValues(onlyIfMaterialized)` then I don't think we need to include a `maybe` or `IfPossible` in the name as this is implied already by the `onlyIf`.

Reply:
I guess it's subjective. Personally, I would prefer to flip it.

Reply:
OK, will flip.